screen/send/details.js

```diff
@@ -609,16 +609,26 @@ const SendDetails = () => {
   const onUseAllPressed = () => {
     ReactNativeHapticFeedback.trigger('notificationWarning');
     Keyboard.dismiss();
-    setAddresses(addrs => {
-      addrs[scrollIndex.current].amount = BitcoinUnit.MAX;
-      addrs[scrollIndex.current].amountSats = BitcoinUnit.MAX;
-      return [...addrs];
-    });
     setUnits(u => {
       u[scrollIndex.current] = BitcoinUnit.BTC;
       return [...u];
     });
-    LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
+    setAddresses(addrs => {
+      const currentBalance = wallet.getBalance();
+      addrs[scrollIndex.current].amountSats = currentBalance;
+      switch (units[scrollIndex.current]) {
+        case BitcoinUnit.SATS:
+          addrs[scrollIndex.current].amount = currency.satoshiToBTC(currentBalance);
+          break;
+        case BitcoinUnit.BTC:
+          addrs[scrollIndex.current].amount = currency.satoshiToBTC(currentBalance);
+          break;
```

Comment:
Why not summarize if it's the same logic?
```ts
        case BitcoinUnit.SATS:
        case BitcoinUnit.BTC:
          addrs[scrollIndex.current].amount = currency.satoshiToBTC(currentBalance);
          break;
```

Reply:
done, thanks!